File: WebAPI/Controllers/FeedbackTypeController.cs
using System.Threading.Tasks;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using ReachUp;

namespace ReachUpWebAPI.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class FeedbackTypeController : ControllerBase
    {
        // OK 
        [Authorize(Roles = "adm, dev")]
        [HttpGet("GetAll")]
        public async Task<IActionResult> GetAll()
        {
           return Ok(await new FeedbackType().GetAll());
        }

        // OK 
        [Authorize(Roles = "adm, dev")]
        [HttpPost]
        public async Task<IActionResult> Post([FromBody] FeedbackType feedbackType)
        {
            if (feedbackType != null)
                return Ok(await feedbackType.Add());
            return BadRequest("Parameters are null");
        }

        // OK 
        [Authorize(Roles = "adm")]
        [HttpPatch]
        public async Task<IActionResult> Patch([FromBody] FeedbackType feedbackType)
        {
            if (feedbackType != null)
                return Ok(await feedbackType.Update());
            return BadRequest("Parameters are null");
        }

        // OK 
        [Authorize(Roles = "adm")]
        [HttpDelete]
        public async Task<IActionResult> Delete(int id)
        {
            if (!string.IsNullOrWhiteSpace(id.ToString()))
                return Ok(await new FeedbackType().Delete(id));
            return BadRequest("Parameters are null");
        }

    }
}

File: WebAdmin/src/Controllers/FeedbackTypeController.js
class FeedbackTypeController {
    api;
    constructor(){
        this.api = new ApiConnection();
    }

    async getAll()
    {
        return await
         this.api.httpAnonymousGet('FeedbackType/GetAll');
    }

    async add(name)
    {
        return await
         this.api.httpAnonymousPost(
             'FeedbackType/',
             {
                name: name
             }
        );
    }

    async update(id, name)
    {
        return await
         this.api.httpAnonymousPatch(
             'FeedbackType/',
             {
                id: id,
                name: name
             }
        );
    }

    async delete(id)
    {
        return await
         this.api.httpAnonymousDelete(`FeedbackType?id=${id}`);
    }
}